### ISSUE TITLE
t/pod/*.t tests not run by harness or make test

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2613]
Migrated from [rt.perl.org#4288](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=4288) (status was 'resolved')

Searchable as RT4288$

**Comment by p5pRT** [comment]
### From @ysth

#### Created by @ysth



Not sure if this is an oversight or not\, but the perl distribution
includes t/pod/\*\.t tests which don't get run by harness or make test\.

Here's a patch&#8203;:
<details><summary>Inline Patch</summary>

```diff
--- t/TEST.orig	Sun Aug 13 12:37:00 2000
+++ t/TEST	Sun Sep 10 22:36:28 2000
@@ -24,7 +24,7 @@
 
 if ($#ARGV == -1) {
     @ARGV = split(/[ \n]/,
-      `echo base/*.t comp/*.t cmd/*.t io/*.t; echo op/*.t pragma/*.t lib/*.t`);
+      `echo base/*.t comp/*.t cmd/*.t io/*.t; echo op/*.t pod/*.t pragma/*.t; echo lib/*.t`);
 }
 
 %infinite = ( 'comp/require.t', 1, 'op/bop.t', 1, 'lib/hostname.t', 1 ); 
--- t/harness.orig	Sun Aug 13 12:36:00 2000
+++ t/harness	Mon Sep 11 15:37:36 2000
@@ -37,7 +37,7 @@
 }
 
 @tests = @ARGV;
-@tests = <base/*.t comp/*.t cmd/*.t io/*.t op/*.t pragma/*.t lib/*.t> unless @tests;
+@tests = <base/*.t comp/*.t cmd/*.t io/*.t op/*.t pod/*.t pragma/*.t lib/*.t> unless @tests;
 
 Test::Harness::runtests @tests;
 exit(0) unless -e "../testcompile";

```
</details>
End of Patch.

But perhaps these should just be moved to t/lib instead?  \(with a pod\- prefix\)

Also note that t/pod/find\.t fails tests 2 and 3\.

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=low

Site configuration information for perl v5.7.0:

Configured by sthoenna at Mon Sep 11 14:25:55 PDT 2000.

Summary of my perl5 (revision 5.0 version 7 subversion 0) configuration:
  Platform:
    osname=os2, osvers=2.30, archname=os2-64int-ld
    uname='os2 efn.org 2 2.30 i386  '
    config_args='-de -Dprefix=d:/perl -Dusedevel -Duse64bitint -Duselongdouble -Aoptimize=-DDEBUGGING'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=define use64bitall=undef uselongdouble=define
  Compiler:
    cc='gcc', ccflags ='-Zomf -Zmt -DDOSISH -DOS2=2 -DEMBED -I. -D_EMX_CRT_REV_=63',
    optimize='-O2 -fomit-frame-pointer -malign-loops=2 -malign-jumps=2 -malign-functions=2 -s -DDEBUGGING',
    cppflags='-Zomf -Zmt -DDOSISH -DOS2=2 -DEMBED -I. -D_EMX_CRT_REV_=63'
    ccversion='', gccversion='2.8.1', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long long', ivsize=8, nvtype='long double', nvsize=12, Off_t='off_t', lseeksize=4
    alignbytes=4, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-Zexe -Zomf -Zmt -Zcrtdll -Zstack 32000'
    libpth=d:/emx/lib d:/emx/lib/mt
    libs=-lsocket -lm -lbsd
    libc=d:/emx/lib/mt/c_import.lib, so=dll, useshrplib=true, libperl=libperl.lib
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-Zdll', lddlflags='-Zdll -Zomf -Zmt -Zcrtdll -s'

Locally applied patches:
    DEVEL7048


@INC for perl v5.7.0:
    lib
    d:/perl/lib/5.7.0/os2-64int-ld
    d:/perl/lib/5.7.0
    d:/perl/lib/site_perl/5.7.0/os2-64int-ld
    d:/perl/lib/site_perl/5.7.0
    d:/perl/lib/site_perl
    .


Environment for perl v5.7.0:
    HOME=d:\home\sthoenna
    LANG=en_us
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR=sthoenna
    PATH=d:\bin;C:\OS2;d:\perl\bin;C:\OS2\SYSTEM;C:\OS2\INSTALL;C:\;C:\OS2\MDOS;C:\OS2\APPS;C:\MMOS2;d:\os2apps\util;d:\DOSAPPS\UTIL;c:\sio;D:\WINDOWS;d:\pdksh;d:\emx\bin;d:\emacs\19.33\bin;d:\ispell
    PERL_BADLANG (unset)
    PERL_SH_DIR=d:\BIN
    SHELL (unset)


```
</details>

**Comment by p5pRT** [comment]
### From @jhi

On Mon\, Sep 11\, 2000 at 05&#8203;:56&#8203;:47PM \-0700\, sthoenna@&#8203;efn\.org wrote&#8203;:
> This is a bug report for perl from sthoenna@&#8203;efn\.org\,
> generated with the help of perlbug 1\.32 running under perl v5\.7\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> Not sure if this is an oversight or not\, but the perl distribution

Oversight\.

> includes t/pod/\*\.t tests which don't get run by harness or make test\.
> But perhaps these should just be moved to t/lib instead?  \(with a pod\- prefix\)

Yes\.

> Also note that t/pod/find\.t fails tests 2 and 3\.

P\-\-\-\-\-\- w\-\-\-\-\-\-\.

**Comment by p5pRT** [comment]
### From @jhi

On Mon\, Sep 11\, 2000 at 05&#8203;:56&#8203;:47PM \-0700\, sthoenna@&#8203;efn\.org wrote&#8203;:
> This is a bug report for perl from sthoenna@&#8203;efn\.org\,
> generated with the help of perlbug 1\.32 running under perl v5\.7\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> Not sure if this is an oversight or not\, but the perl distribution
> includes t/pod/\*\.t tests which don't get run by harness or make test\.
> 
> Here's a patch&#8203;:

Hmmm\.  With 5\.7\.0\+ the patch applied okay but the result fell over
hard under 'make test'\.

**Comment by p5pRT** [comment]
### From @ysth

In article \<20000912101017\.A26201@&#8203;chaos\.wustl\.edu>\,
Jarkko Hietaniemi \<jhi@&#8203;iki\.fi> wrote&#8203;:
> On Mon\, Sep 11\, 2000 at 05&#8203;:56&#8203;:47PM \-0700\, sthoenna@&#8203;efn\.org wrote&#8203;:
> > This is a bug report for perl from sthoenna@&#8203;efn\.org\,
> > generated with the help of perlbug 1\.32 running under perl v5\.7\.0\.
> > 
> > 
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > \[Please enter your report here\]
> > 
> > Not sure if this is an oversight or not\, but the perl distribution
> > includes t/pod/\*\.t tests which don't get run by harness or make test\.
> > 
> > Here's a patch&#8203;:
> 
> Hmmm\.  With 5\.7\.0\+ the patch applied okay but the result fell over
> hard under 'make test'\.

Works ok here patched up through 7063\, except for pod/find\.t failing
\(which seems to be because a\) it expects to run in the directory above
lib/\, not in t/\, and b\) it doesn't expect there to be any modules
there other than the Pod ones\)\.  What do you mean by "fell over hard"?

**Comment by p5pRT** [comment]
### From @jhi

On Tue\, Sep 12\, 2000 at 04&#8203;:21&#8203;:35PM \-0700\, Yitzchak Scott\-Thoennes wrote&#8203;:
> In article \<20000912101017\.A26201@&#8203;chaos\.wustl\.edu>\,
> Jarkko Hietaniemi \<jhi@&#8203;iki\.fi> wrote&#8203;:
> > On Mon\, Sep 11\, 2000 at 05&#8203;:56&#8203;:47PM \-0700\, sthoenna@&#8203;efn\.org wrote&#8203;:
> > > This is a bug report for perl from sthoenna@&#8203;efn\.org\,
> > > generated with the help of perlbug 1\.32 running under perl v5\.7\.0\.
> > > 
> > > 
> > > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > > \[Please enter your report here\]
> > > 
> > > Not sure if this is an oversight or not\, but the perl distribution
> > > includes t/pod/\*\.t tests which don't get run by harness or make test\.
> > > 
> > > Here's a patch&#8203;:
> > 
> > Hmmm\.  With 5\.7\.0\+ the patch applied okay but the result fell over
> > hard under 'make test'\.
> 
> Works ok here patched up through 7063\, except for pod/find\.t failing
> \(which seems to be because a\) it expects to run in the directory above
> lib/\, not in t/\, and b\) it doesn't expect there to be any modules
> there other than the Pod ones\)\.  What do you mean by "fell over hard"?

All the 15 pod tests failed for me under "make test"\.  First\, because
they could not find File&#8203;::Basename\.  I fixed that by adding the
customary "chdir 't' if \-d 't';@&#8203;INC='\.\./lib';" mantra\.  Then they
couldn't find the testp2pt\.pl\.  I fixed that by push @&#8203;INC\, "pod"\.
Then the testp2pt\.pl failed since it couldn't find t/something\, or was
it pod/something\.  Anyway\, then I grew too disgusted and reverted the
patch\.
